Task3/src/main/java/Controller.java

```diff
@@ -67,6 +41,11 @@ public void shutElevator() {
     public void moveElevator() {
         lock.lock();
         int currentFloor = elevator.getCurrentFloor();
+        if (currentFloor == elevator.getTopFloor()) {
+            elevator.setMovementDirection(MovementDirection.DOWN);
+        } else if (currentFloor == 0) {
+            elevator.setMovementDirection(MovementDirection.UP);
+        }
         int nextFloor;
         switch (elevator.getMovementDirection()) {
             case UP -> nextFloor = currentFloor + 1;
```

Comment:
You can place information about inrement to MovementDirection enum. It'll allow you to avoid switch statement.
It should look like:
```
nextFloor + UP.incrementFloor()
```